compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs

```diff
@@ -1055,7 +1056,15 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     } else {
                         "".to_string()
                     };
-                    labels.push((provided_span, format!("unexpected argument{provided_ty_name}")));
+                    let ord = if provided_arg_tys.len() == 1 {
+                        "".to_string()
+                    } else {
+                        format!("#{} ", arg_idx.as_usize() + 1)
+                    };
+                    labels.push((
+                        provided_span,
+                        format!("unexpected {ord}argument{provided_ty_name}"),
```

Comment:
```suggestion
                        format!("unexpected argument{idx}{provided_ty_name}"),
```

In English, \*“#1 argument” (\*“number one argument”) is ungrammatical, it's “argument #1” (“argument number one”).